packages/app-extensions/src/form/reduxForm.js

```diff
@@ -62,19 +63,22 @@ export const transformFieldNameBack = fieldName => (fieldName.replace(/--/g, '.'
 
 export const entityToFormValues = entity => {
   if (!entity || !entity.paths) return {}
-  const result = {}
-  const paths = entity.paths
+
+  const result = {
+    [versionField]: entity.version
+  }
 
   Object.keys(entity.paths).forEach(key => {
-    const field = paths[key]
-    if (field.type === 'entity' || field.type === 'entity-list' || field.type === 'display-expression') {
-      result[transformFieldName(key)] = field.value
-    } else {
-      result[transformFieldName(key)] = field.value ? field.value.value : null
-    }
+    const field = entity.paths[key]
+    const type = field.type
```

Comment:
(there's only one usage of the `type` variable and it's only barely shorter than `field.type` -> I personally would inline the variable on line 75)